pipeline/coordinator.go

```diff
@@ -254,18 +302,71 @@ func (c *Coordinator) startOneUploadJob(p UploadJobPayload, handler Handler, for
 		state:              "segmenting",
 		transcodedSegments: 0,
 		catalystRegion:     os.Getenv("MY_REGION"),
+		UnifiedUpload:      c.TransferURL != nil,
 	}
 	si.ReportProgress(clients.TranscodeStatusPreparing, 0)
 
 	c.Jobs.Store(streamName, si)
 	log.Log(si.RequestID, "Wrote to jobs cache")
 
 	c.runHandlerAsync(si, func() (*HandlerOutput, error) {
+		if c.TransferURL == nil {
+			log.Log(si.RequestID, "no unified transfer")
+		} else {
+			targetDir := getTargetDir(si.RequestID, si.TargetURL)
```

Comment:
I think all the logic from here on in this `else` case would be nicer to move away to a `CopyAndProbe` fcn?